### PR TITLE
Split out memory component to update on add and tidied up some auth

### PR DESCRIPTION
**Pull request by ChangoFoster**
* Made the memory component work for now until/if we add redux so it updates the list when you add something new
* Apparently eval is unsafe so I've just swapped that for a library (js-cookie) and JSON.parse
* Added in the verifytoken middleware to the me endpoint
* I removed the cookie.js file as I don't think that's used any more?

---
This change is [Reviewable](https://reviewable.io/reviews/jamestwisleton/musicjournal/9)

### CODE REVIEW

File: music-journal/components/Memory.js [hunk 1]
```diff
@@ -1,12 +1,46 @@
...
+    }, []);
+
+    const handleSubmit = async (event) => {
+        event.preventDefault()
+
+        try {
+            const newMemory = await addMemory({ song, text }, props.token)
+            setSong('')
+            setText('')
+            setMemories({ ...memories, ...newMemory })
```
Comment (JamesTwisleton) [inline music-journal/components/Memory.js:28]: so this essentially appends the new memory to the existing list of memories, then sets the memories to be the new list?

File: music-journal/pages/api/list-memories.js [hunk 2]
```diff
@@ -26,13 +26,9 @@ async function handler(req, res) {
...
 
-        let memories = []
-
         const memoriesRef = await database.ref(`/memory/${user.uid}`);
-
-        await memoriesRef.orderByValue().once("value", snapshot => {
-            snapshot.forEach(data => { memories.push(data.val()) })
-        })
+        const memoriesSnapshot = await memoriesRef.orderByValue().once("value")
+        const memories = Object.assign({}, memoriesSnapshot.val())
```
Comment (JamesTwisleton) [inline music-journal/pages/api/list-memories.js:31]: ah yeah this is what you linked me!

File: music-journal/pages/api/save-memory.js [hunk 3]
```diff
@@ -14,27 +14,31 @@ const cors = initMiddleware(
...
 
 async function handler(req, res) {
-    const messageUuid = uuidv4();
     await cors(req, res)
     await verifyTokenMiddleware(req, res);
 
-    if (!req.body.song || !req.body.memoryText) {
+    const messageUuid = uuidv4();
+    const user = auth().currentUser;
+    const { song, text } = req.body
```
Comment (JamesTwisleton) [inline music-journal/pages/api/save-memory.js:22]: awesome!

File: music-journal/utils/auth.js [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-}
-export default async function auth(cookie) {
-    const firebaseToken = deserialize(cookie.replace('__session=','')).firebaseToken;
-    const { data: { user }, status } = await fetcher(`/api/me?token=${firebaseToken}`);
-
-    if(status !== 200) {
-        return false;
+export default async function auth() {
+    try {
+        const { firebaseToken } = JSON.parse(Cookies.get('__session'))
```
Comment (JamesTwisleton) [inline music-journal/utils/auth.js:6]: I'm so glad this is working, during the `cookie trials` I was thrashing with js-cookie, react-cookie and others trying to do this

File: music-journal/utils/withAuth.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ const withAuth = (Component) => {
         async componentDidMount() {
             if (document.cookie) {
                 try {
-                    const [token, user] = await auth(document.cookie);
+                    const [token, user] = await auth();
```
Comment (JamesTwisleton) [inline music-journal/utils/withAuth.js:18]: so much better to have it encapsulated within auth()

**Review by JamesTwisleton — APPROVED** [review]
👏

**Comment by JamesTwisleton** [comment]
This is deployed to https://musicjournal-c3d3e.web.app/ and works like a charm :) feel free to merge it whenever G bloody G!

File: music-journal/components/Memory.js (hunk 1 above)
Comment (ChangoFoster) [reply music-journal/components/Memory.js:28]: Yep, that's it

File: music-journal/pages/api/list-memories.js (hunk 2 above)
Comment (ChangoFoster) [reply music-journal/pages/api/list-memories.js:31]: Yeah I much prefer that style to the callbacks in callbacks

File: music-journal/utils/auth.js (hunk 4 above)
Comment (ChangoFoster) [reply music-journal/utils/auth.js:6]: Well I never would have figured it out at all! Was easy to tidy up when you had it all working

**Pull request merged**